### PR TITLE
sci-libs/clblast: version bump to 1.5.1

### PR DESCRIPTION
**Pull request by epsilon-0**
adds lib{,c}blas.so.3 replacement using eselect-blas

Closes: https://bugs.gentoo.org/747142

Package-Manager: Portage-3.0.8, Repoman-3.0.1
Signed-off-by: Aisha Tammy <gentoo@aisha.cc>

### CODE REVIEW

**Comment by epsilon-0** [comment]
All tests are passing.

File: sci-libs/clblast/clblast-1.5.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	# no forced optimisation, libdir
+	sed -e 's/-O3//g' -e 's/-O2//g' \
+		-e 's/DESTINATION lib/DESTINATION ${CMAKE_INSTALL_LIBDIR}/g' \
+		-i CMakeLists.txt || die
+	cmake_src_prepare
+}
+
+src_configure() {
+	mycmakeargs+=(
+		-DBUILD_SHARED_LIBS=ON
```
Comment (SoapGentoo) [inline sci-libs/clblast/clblast-1.5.1.ebuild:41]: try removing this, not needed on EAPI 7 anymore

File: sci-libs/clblast/clblast-1.5.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		-DVERBOSE=ON
+		-DSAMPLES=OFF
+		-DTUNERS=ON
+		-DOPENCL=ON
+		-DCUDA=OFF
+		-DNETLIB=ON
+		-DNETLIB_PERSISTENT_OPENCL=$(usex persistent)
+		-DCLIENTS="$(usex client)"
+		-DTESTS="$(usex test)"
+		"${EXTRA_ECONF[@]}"
```
Comment (SoapGentoo) [inline sci-libs/clblast/clblast-1.5.1.ebuild:51]: this is only for `econf`, `MYCMAKEARGS` is the cmake analogue, and isn't required explicitly

File: sci-libs/clblast/clblast-1.5.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	cmake_src_test alltests
+}
+
+src_install() {
+	cmake_src_install
+	dodoc README.md CONTRIBUTING.md CHANGELOG
+	use doc && dodoc -r doc
+	if use examples; then
+		docinto /usr/share/doc/${PF}/examples
+		dodoc -r samples/*
```
Comment (SoapGentoo) [inline sci-libs/clblast/clblast-1.5.1.ebuild:72]: `dodoc -r samples/.` (try and avoid globs is possible)

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sci-libs/clblast/clblast-1.5.1.ebuild (hunk 1 above)
Comment (epsilon-0) [reply sci-libs/clblast/clblast-1.5.1.ebuild]: done

File: sci-libs/clblast/clblast-1.5.1.ebuild (hunk 2 above)
Comment (epsilon-0) [reply sci-libs/clblast/clblast-1.5.1.ebuild]: done

File: sci-libs/clblast/clblast-1.5.1.ebuild (hunk 3 above)
Comment (epsilon-0) [reply sci-libs/clblast/clblast-1.5.1.ebuild]: done

**Comment by heroxbd** [comment]
It is installing a lot of tests into /, such as,

```
   /usr/bin/clblast_test_xnrm2                                                                            
   /usr/bin/clblast_test_xamax                                                                            
   /usr/bin/clblast_test_xtrmv                                                                            
   /usr/bin/clblast_test_xasum                                                                            
   /usr/bin/clblast_test_xtpmv                                                                            
   /usr/bin/clblast_test_xgemv                                                                            
   /usr/bin/clblast_test_xgbmv                                                                            
   /usr/bin/clblast_test_xsyr                                                                             
   /usr/bin/clblast_test_xsymv                                                                            
   /usr/bin/clblast_test_xgerc                                                                            
   /usr/bin/clblast_test_xhemv                                                                            
   /usr/bin/clblast_test_xhbmv                                                                            
   /usr/bin/clblast_test_xhpmv                                                                            
   /usr/bin/clblast_test_xsbmv                                                                            
   /usr/bin/clblast_test_xtbmv                                                                            
   /usr/bin/clblast_test_xspmv                                                                            
   /usr/bin/clblast_test_xspr                                                                             
   /usr/bin/clblast_test_xgeru                                                                            
   /usr/bin/clblast_test_xher2k                                                                           
   /usr/bin/clblast_test_xgemm                                                                            
   /usr/bin/clblast_test_xtrsv                                                                            
   /usr/bin/clblast_test_xger                                                                             
   /usr/bin/clblast_test_xhemm                                                                            
   /usr/bin/clblast_test_xtrmm                                                                                                                                                                                      
   /usr/bin/clblast_test_xhpr                                                                             
   /usr/bin/clblast_test_xher                                                                                                                                                                                       
   /usr/bin/clblast_test_xherk                                                                            
   /usr/bin/clblast_test_xsyr2k  
```

**Comment by epsilon-0** [comment]
> It is installing a lot of tests into /, such as,
> 
> ```
>    /usr/bin/clblast_test_xnrm2                                                                            
>    /usr/bin/clblast_test_xamax                                                                            
>    /usr/bin/clblast_test_xtrmv                                                                            
>    /usr/bin/clblast_test_xasum                                                                            
>    /usr/bin/clblast_test_xtpmv                                                                            
>    /usr/bin/clblast_test_xgemv                                                                            
>    /usr/bin/clblast_test_xgbmv                                                                            
>    /usr/bin/clblast_test_xsyr                                                                             
>    /usr/bin/clblast_test_xsymv                                                                            
>    /usr/bin/clblast_test_xgerc                                                                            
>    /usr/bin/clblast_test_xhemv                                                                            
>    /usr/bin/clblast_test_xhbmv                                                                            
>    /usr/bin/clblast_test_xhpmv                                                                            
>    /usr/bin/clblast_test_xsbmv                                                                            
>    /usr/bin/clblast_test_xtbmv                                                                            
>    /usr/bin/clblast_test_xspmv                                                                            
>    /usr/bin/clblast_test_xspr                                                                             
>    /usr/bin/clblast_test_xgeru                                                                            
>    /usr/bin/clblast_test_xher2k                                                                           
>    /usr/bin/clblast_test_xgemm                                                                            
>    /usr/bin/clblast_test_xtrsv                                                                            
>    /usr/bin/clblast_test_xger                                                                             
>    /usr/bin/clblast_test_xhemm                                                                            
>    /usr/bin/clblast_test_xtrmm                                                                                                                                                                                      
>    /usr/bin/clblast_test_xhpr                                                                             
>    /usr/bin/clblast_test_xher                                                                                                                                                                                       
>    /usr/bin/clblast_test_xherk                                                                            
>    /usr/bin/clblast_test_xsyr2k  
> ```

This looks weird :confused:, I have no inclination why it's doing that..
I will poke around and see.

File: sci-libs/clblast/clblast-1.5.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+src_prepare() {
+	# no forced optimisation, libdir
+	sed -e 's/-O3//g' -e 's/-O2//g' \
+		-e 's/DESTINATION lib/DESTINATION ${CMAKE_INSTALL_LIBDIR}/g' \
+		-i CMakeLists.txt || die
+	cmake_src_prepare
+}
+
+src_configure() {
+	mycmakeargs+=(
```
Comment (a17r) [inline sci-libs/clblast/clblast-1.5.1.ebuild:40]: `local mycmakeargs=(`

**Review by a17r — CHANGES_REQUESTED** [review]

**Comment by a17r** [comment]
> > It is installing a lot of tests into /, such as,
> > ...
> This looks weird , I have no inclination why it's doing that..
> I will poke around and see.

The reason for that is scattered over the root CMakeLists.txt:

starting with line 444:
```
  if(OPENCL)
    foreach(ROUTINE_TUNER ${ROUTINE_TUNERS})
      add_executable(clblast_tuner_routine_${ROUTINE_TUNER} ${TUNERS_COMMON} src/tuning/routines/${ROUTINE_TUNER}.cpp test/test_utilities.cpp)
      target_link_libraries(clblast_tuner_routine_${ROUTINE_TUNER} clblast)
      target_include_directories(clblast_tuner_routine_${ROUTINE_TUNER} PUBLIC $<TARGET_PROPERTY:clblast,INTERFACE_INCLUDE_DIRECTORIES> ${API_INCLUDE_DIRS} ${clblast_>
      install(TARGETS clblast_tuner_routine_${ROUTINE_TUNER} DESTINATION bin)
    endforeach()
  endif()
```

and later line 549:
```
  # Compiles the performance-tests
  foreach(ROUTINE ${LEVEL1_ROUTINES})
    add_executable(clblast_client_${ROUTINE} ${CLIENTS_COMMON}
                   test/performance/routines/level1/${ROUTINE}.cpp
                   test/routines/level1/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${LEVEL2_ROUTINES})
    add_executable(clblast_client_${ROUTINE} ${CLIENTS_COMMON}
                   test/performance/routines/level2/${ROUTINE}.cpp
                   test/routines/level2/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${LEVEL3_ROUTINES})
    add_executable(clblast_client_${ROUTINE} ${CLIENTS_COMMON}
                   test/performance/routines/level3/${ROUTINE}.cpp
                   test/routines/level3/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${LEVELX_ROUTINES})
    add_executable(clblast_client_${ROUTINE} ${CLIENTS_COMMON}
                   test/performance/routines/levelx/${ROUTINE}.cpp
                   test/routines/levelx/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${ROUTINES})
    target_link_libraries(clblast_client_${ROUTINE} clblast ${REF_LIBRARIES} ${API_LIBRARIES})
    target_include_directories(clblast_client_${ROUTINE} PUBLIC ${clblast_SOURCE_DIR} ${REF_INCLUDES})
    install(TARGETS clblast_client_${ROUTINE} DESTINATION bin)
  endforeach()

endif()
```

and finally line 601:
```
  # Compiles the correctness-tests
  foreach(ROUTINE ${LEVEL1_ROUTINES})
    add_executable(clblast_test_${ROUTINE} ${TESTS_COMMON}
                   test/correctness/routines/level1/${ROUTINE}.cpp
                   test/routines/level1/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${LEVEL2_ROUTINES})
    add_executable(clblast_test_${ROUTINE} ${TESTS_COMMON}
                   test/correctness/routines/level2/${ROUTINE}.cpp
                   test/routines/level2/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${LEVEL3_ROUTINES})
    add_executable(clblast_test_${ROUTINE} ${TESTS_COMMON}
                   test/correctness/routines/level3/${ROUTINE}.cpp
                   test/routines/level3/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${LEVELX_ROUTINES})
    add_executable(clblast_test_${ROUTINE} ${TESTS_COMMON}
                   test/correctness/routines/levelx/${ROUTINE}.cpp
                   test/routines/levelx/${ROUTINE}.hpp)
  endforeach()
  foreach(ROUTINE ${ROUTINES})
    target_link_libraries(clblast_test_${ROUTINE} clblast ${REF_LIBRARIES} ${API_LIBRARIES})
    install(TARGETS clblast_test_${ROUTINE} DESTINATION bin)
    target_include_directories(clblast_test_${ROUTINE} PUBLIC ${clblast_SOURCE_DIR} ${REF_INCLUDES})
    add_test(clblast_test_${ROUTINE} clblast_test_${ROUTINE})
  endforeach()
```

Optional build of tests using `BUILD_TESTING` from CMake `CTest` module, and another option to separately install tests, should be patched and sent upstream.

**Comment by a17r** [comment]
Meanwhile, 1.5.2 was released without relevant fixes for our issue, do take note of an additional patch after release though: https://github.com/CNugteren/CLBlast/commit/3d3492646c511e7617c99a59a3def93f22189d72

**Comment by arkamar** [comment]
Version 1.5.2 is in the tree since commit 870b2648439a - `sci-libs/clblast: add version 1.5.2`. The version bump here is obsolete but I don't know about https://bugs.gentoo.org/747142 fix.

**Pull request closed without merging**